client/AccountsClient.ts

```diff
@@ -62,95 +58,130 @@ export default class AccountsClient {
         return this._redirectClient.init();
     }
 
-    public on(
-        command: RequestType,
-        resolve: (result: RpcResult, state: any) => any,
-        reject?: (error: Error, state: any) => any,
+    public on<T extends RequestType>(
+        command: T,
+        resolve: (result: ResultByRequestType<T>, state: any) => void,
+        reject?: (error: Error, state: any) => void,
     ) {
         this._redirectClient.onResponse(command,
             // State is always an object containing at least the __command property
-            (result: RpcResult, rpcId, state) => resolve(result, state),
-            (error: Error, rpcId, state) => reject && reject(error, state),
+            (result: ResultByRequestType<T>, rpcId, state) => resolve(result, state),
+            (error: Error, rpcId, state) => {
+                if (reject) {
```

Comment:
+1 for removing the `reject && ...`.
I'd rather use `if (!reject) return;` though.